##**Урок 8. Дисперсионный анализ. Логистическая регрессия**

**1.Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.**

In [1]:
import numpy as np 
from scipy.stats import f_oneway

In [2]:
football = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hockey = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
weightlifter = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [3]:
football_mean = football.mean()
football_mean

179.125

In [4]:
hockey_mean = hockey.mean()
hockey_mean

178.66666666666666

In [5]:
weightlifter_mean = weightlifter.mean()
weightlifter_mean

172.72727272727272

In [6]:
Y = np.hstack((football, hockey, weightlifter))
Y_mean = Y.mean()
Y_mean

176.46428571428572

In [7]:
S2_F = (football_mean - Y_mean)**2 * len(football) + (hockey_mean - Y_mean)**2  * len(hockey) + (weightlifter_mean - Y_mean)**2 * len(weightlifter)
S2_F

253.9074675324678

In [8]:

S2_res = ((football - football_mean)**2).sum() + ((hockey - hockey_mean)**2).sum() + ((weightlifter - weightlifter_mean)**2).sum()
S2_res

577.0568181818182

In [9]:

q2_F = S2_F / (3 - 1)
q2_F

126.9537337662339

In [10]:
q2_res = S2_res / (len(Y) - 3)
q2_res

23.08227272727273

In [11]:
F_H = q2_F / q2_res
F_H

5.500053450812598

$F_H = \frac{\sigma^2_F}{\sigma^2_{\text{ост}}} = 5.500$ 

$F_{\text{крит.}} = 3.38 \quad \text{при }\alpha = 0.05$

$5.500 > 3.38$

In [12]:

f_oneway(football, hockey, weightlifter)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Различия между группами статистически значимы на уровне значимости 0.05.